cmd/agent/app/reporter/flags.go

```diff
@@ -46,9 +44,16 @@ type Options struct {
 	AgentTags    map[string]string
 }
 
+// Flags holds configuration for AddFlags.
+type Flags struct {
+	// Reporters which area added to reporter type help message
+	Reporters []string
+}
+
 // AddFlags adds flags for Options.
-func AddFlags(flags *flag.FlagSet) {
-	flags.String(reporterType, string(GRPC), fmt.Sprintf("Reporter type to use e.g. %s, %s[%s]", string(GRPC), string(TCHANNEL), "NOTE: Deprecated since 1.16"))
+func (f Flags) AddFlags(flags *flag.FlagSet) {
+	reps := append([]string{string(GRPC)}, f.Reporters...)
+	flags.String(reporterType, string(GRPC), fmt.Sprintf("Reporter type to use e.g. %s", reps))
```

Comment:
if all this complication just to include tchannel in the possible reporter types in the help string, I would say it's not worth it.

Reply:
Yes just for the help message, I am removing it